src/material/tree/testing/tree-harness.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {
+  MatNestedTreeNodeHarness,
+  MatTreeNodeHarness,
+  TreeHarnessFilters,
+  TreeNodeHarnessFilters
+} from '@angular/material/tree/testing';
+
+/** Harness for interacting with a standard mat-tree in tests. */
+export class MatTreeHarness extends ComponentHarness {
```

Comment:
We might want to add an additional method to get the text as some type of structured data. I think @andrewseguin did something similar for table

Reply:
I could see (in a follow-up) adding a harness method here like `prettyPrint()` that just prints out the text of the tree in a human-readable form for debugging. Table doesn't have something like this, but maybe it could. 

The table harness (implemented by Kristiyan) does have convenience method for getting cell text strings, which we could do here too (also in a follow-up)